Paola leave centre (#23)
coecms/cms-wiki: Wiki/starting/starting-leaving.md

# Leaving CLEX guidelines

> When leaving the Centre it is important to document and organise the data and code you are leaving beyond. This is both to ensure that unecessary files are removed and storage freed up in shared projects and that anything of value to your future self and/or others is properly stored and documented. 

Reviewer: ```suggestion
When leaving the Centre it is important to document and organise the data and code you are leaving behind. This arrangement ensures that unecessary files are removed, storage is freed up in shared projects, and anything of future value to yourself or others is properly stored and documented. 
```

These guidelines are provided to cover the most common workflow. 

You should not own any data on disk when you leave or shortly after you leave
Any data you own on tape should be accessible for read and write by other people

> ## The Exit Email

Reviewer: ```suggestion
## Your Exit Email Notification
```

> About 2 months before you leave, you should receive the following email:

Reviewer: ```suggestion
About 2 months prior to your departure, you should receive the following email:
```

> 

Reviewer: I am not suggesting any changes to the email template as I am not sure you want me to do that, but the email text can definitely be improved.

Author: Stephen sends that, so unless there are major changes, it's not worth doing anything, at least for the moment.

Dear xxxx,

Given that your contract is due to end in 2 months time, we ask that you start considering how data in your accounts will be managed in future. Specifically:
- please delete anything that is not required
- please send an email to the CMS team (copied on this email), detailing the nature of what will remain. This should outline major directories and explain:
	- what they contain 
	- why they’re important to keep
	- what is required to access them - which repository / scripts?
	- if this data is stored anywhere else
	- when can they be deleted (date)
	- where this data should be stored when your account is shut down (please consult with a CLEX CI)
	- who in CLEX should have access to this data?
	- please also let the CMS team if compute access will be required after your contract ends
 - please reply to both the admin and CMS teams with future contact details when you have them

The CMS team can help with tools and techniques to help you manage your data. Please understand that storage is extremely limited, so that a lack of response to the questions above will likely result in your data being archived as soon as your contract ends.

Regards,

The CLEX Admin Team
```

> While you will receive the email above to remind you what needs to be done, sorting through your files might take longer than you think and it is important to be organised at any stage of your project. So please consider running the suggested actions at different stages of your project whne appropriate.

Reviewer: ```suggestion
Although you will receive the above email reminder of what needs to be done, going through all of your files might take longer than anticipated, thus it is important to remain organised at every stage of your project. Therefore, please consider performing the suggested actions at various stages of your project as appropriate.
```

You might need to discuss the following questions with your supervisor:

* Will you require access to the files after you leave (e.g. for a paper review)?

> * What files will be useful to others? Should they be published?

Reviewer: ```suggestion
* What files could be useful to others? Should they be published?
```

* What files won't be useful to others? Should they be archived or deleted?

```{note}
If you need specific advice on how to actually transfer or archive your files, you can always ask for assistance to our helpdesk: cws_help@nci.org.au.
```

## What to do

> While what exactly should be kept, removed or published will depend on your specific project and working group. There are some principles that can be applied to any situation.

Reviewer: ```suggestion
While what to keep, remove, or publish, will depend on your specific project and working group, some universal principles can be applied to any situation.
```

> ### To keep, always

Reviewer: ```suggestion
### Always Keep
```

**Codes**

> No matter what the details of your projects are you should always keep the code you used. This means keeping a reference to the exact version and document how you used codes developed by others (e.g. climate models). If you modified them in any way this should also be documented. 

Reviewer: ```suggestion
Regardless of your project details, always keep the code you used with a reference to its exact version, and document your usage of codes developed by others (e.g., climate models). Document also any modifications you made to these codes. 
```

> If you modfication are extensive or you have written the code yourself then you should make it available via GitHub and publish it (Zenodo is the most indicated for this).

Reviewer: ```suggestion
If your modifications are extensive or you wrote the code yourself, you should make the code available via GitHub and publish it (preferably on Zenodo ).
```

```{note}
It is absolutely fine to create a repository per project or paper with all your codes in. You can then use the README file from the GitHub repository to explain how to reproduce your results. Don't forget to clearly reference everything one might need in addition to this repository. Or you can have a repository per code especially if you envision you'll reuse the same code for other work.
```

> **Configuration files for running the codes and some input files**

Reviewer: ```suggestion
**Configuration files and certain input files**
```

Author: Maybe we should rename this completely to "Files needed to run the code" or something similar

Author: done

> In addition to the codes themselves, you need to keep everything that enables someone to run the codes in the same way you have done so. Usually, the most complicated configurations are for climate models. Some climate models will save your configurations in version control repositories (e.g. UM, ACCESS, ACCESS-OM2), in which case you simply need to keep the information on how to retrieve these configurations. Some models don't save your configurations and you need to do it yourself.

Reviewer: ```suggestion
In addition to the codes, keep everything required to run them the same way you did. Usually, the most complicated configurations are for climate models. Some climate models will save your configurations in version control repositories (e.g. UM, ACCESS, ACCESS-OM2), in which case you simply need to keep the information on how to retrieve these configurations. Some models don't save your configurations and you need to do it yourself.
```

> For the input files, some inputs are published data in which case you need to keep the reference to this data (including the version). If you have written several codes, the output of a piece of code will be the input of the next piece of code, in which case you do not necessarily need to keep that data,  but you need to describe your workflow.

Reviewer: ```suggestion
For input files, some are published data, in which case you need to keep the reference to this data (including the version). If you have written several codes, the output of one will be the input to the next, in which case you do not necessarily need to keep that data, but you should describe your workflow.
```

**Workflow**

> This can be a tricky one, as there is not a one-size-fits-all format to save this information. It is fine to write a README file and archive it with other files from the project that need archiving. You can also have a special Github repository just for this README file, or a repository for all your projects with READMEs for each project. Whatever format you choose, it is important for this information to be publicly available (unless your project was restricted) and not a personal note.

Reviewer: ```suggestion
Workflows can be a tricky as there is no one-size-fits-all format to save such information. You could write a README file and archive it with other project files that need archiving, or have a special GitHub repository just for this README file, or a repository for all your projects with READMEs for each project. Whatever you choose, this information must be publicly available (unless your project was restricted) and not a personal note.
```

Author: remove "a" in "Workflows can be a tricky"

Reviewer: Done

> This description should clearly describe step by step what someone should do to reproduce your work.

Reviewer: ```suggestion

```

Author: If I approve this it will show "I have appended this line to the previous paragraph" rather than the intended change

Reviewer: Sorry

Author: done

```{warning}

> Information on your PhD or published paper has to be kept for at least 5 years. The time requirements differ slightly depending on institutions and funding bodies.

Reviewer: ```suggestion
Information on your PhD or published papers must be kept for at least 5 years. The time requirements differ slightly depending on institutions and funding bodies.
```

```

> ### To delete, always

Reviewer: ```suggestion
### Always delete
```

> You do not need to keep files that are not necessary to reproduce your work:

Reviewer: ```suggestion
No need to keep files not necessary to reproduce your work:
```

* log files

> * failed experiments

Reviewer: ```suggestion
* failed experiments files
```

> * temporary files such as created from successive cdo/nco commands.

Reviewer: ```suggestion
* temporary files such as those created from successive cdo/nco commands.
```

```{tip}

> Occasionaly log files might have some information which is relevant to the workflow, still just keeping them makes this information virtually inaccessible to anyone else. The right thing to do in this case is to extract what is useful and add it to the workflow documentation, then getting rid of the files.

Reviewer: ```suggestion
Occasionally log files might contain information relevant to the workflow, but by keeping them this information becomes inaccessible to others. Accordingly, in this case extract what is useful, add it to the workflow documentation, and delete the files.
```

Author: I agree that "but by keeping them this information becomes inaccessible to others." is cleaner but it loses the meaning in this context. so maybe "Occasionally log files might contain information relevant to the workflow, but this is hard to access from the files themselves." or something similar

```

> ### To keep or to delete?

Reviewer: ```suggestion
### keep or delete?
```

> Some of the data is harder to categorise, and need some careful consideration to decide if it needs to be kept or deleted. Neglecting to do so might result in the deletion of useful data or waste of storage

Reviewer: ```suggestion
Some data is harder to categorise, and deciding whether to keep it or delete it requires careful consideration to avoid. the deletion of useful data or wasting storage space.
```

**Model outputs**

> These are usually reproducible, even if it might not be possible to get bitwise reproducibility if underlying libraries change or the machine you have used is retired. 

Reviewer: ```suggestion
These are usually reproducible, although bitwise reproducibility may not be possible if underlying libraries change or the machine you have used is retired. 
```

> Such a decision should be weighed against the cost, in money and time, of reproducing your work.

Reviewer: ```suggestion
The decision to keep or delete model outputs must be weighed against the cost, in money and time, of reproducing your work.
```

> On the other side storing the data can also be expensive, as the model output are usually large,  and of little use if not done properly.

Reviewer: ```suggestion
On the other hand, data storage can also be expensive, as model outputs are usually large.
```

> That is where you need to discuss with your supervisor, and possibly other collaborators, as the answer might vary depending on how likely someone else might find this data useful.
> As model output includes raw output, restart files and post-processed output, they should all be treated differently. usually if keeping raw output this is stored on tape, it is a bit harder to retrieve but tape is less expensive than other storage. Only some of the restart files are kept (if any). Ideally, post-processed should be published if it has been used for publication or if it's going to be shared with others. 

Reviewer: ```suggestion
Discuss this matter with your supervisor, and possibly other collaborators, as the answer might vary depending on how likely someone else might find this data useful.
As model output includes raw output, restart files and post-processed output, they should all be treated differently. usually if keeping raw output this is stored on tape, it is a bit harder to retrieve but tape is less expensive than other storage. Only some of the restart files are kept (if any). Ideally, post-processed should be published if it has been used for publication or if it's going to be shared with others. 
```

Author: Only part of this has been committed ?? I fixed it in my branch

**Output of lengthy processing**

>  It might feel necessary to keep those files but we would argue it is usually not worth the cost of the storage unless there is a clear indication they will be used again soon. The first reason is that with current modern programming techniques a lot of lengthy processing can be shortened significantly. The second reason is the very real cost of the storage is not worth the very hypothetical time saved in the future.

Reviewer: ```suggestion
It is tempting to keep those files but usually not worth the storage cost unless it is clear they will be reused soon, as with current modern programming techniques a lot of lengthy processing can be significantly shortened, and the very real storage cost is not worth any hypothetical time savings in the future.
```

## Publishing data and code

> You should publish your files if the files are frequently used by several other persons. Ideally, that data should be published as soon as its usefulness to others is clear, not just when you leave.

Reviewer: ```suggestion
Publish your files if they are frequently used by others, ideally as soon as their usefulness to others is evident, not just when you leave.
```

> If your files are at NCI, that is all you need to do. If your files are held at your institution, you may need to make sure there is a copy that is accessible by others and owned by someone who is likely to stay for years to come. This might depend on your institution data services.

Reviewer: ```suggestion
If your files are at NCI, there is nothing else to. If your files are held at your institution, you may need to make sure there is a copy accessible to others and owned by someone likely to stay for years to come, all subject to your institution data services.
```

**Sharing files without publication**

Some files might be useful to a small cluster of people but not be worth publishing. In that case, you simply need to change the ownership of the files: get the files to be owned by someone who stays after you.

> If the files are at NCI, you can not change the files' ownership. You and the Lead CI of the project owning the files will need to contact help@nci.org.au so NCI staff can do it for you.

Reviewer: ```suggestion
You cannot change the ownership of files that reside at NCI. You and the Lead CI of the project owning the files need to contact help@nci.org.au so NCI staff can do it for you.
```

```{warning}

> Often researchers shared data with others without publishing it even when this data is used for new publication, this should be avoided at any cost. It does create an issue for the creator of the data who won't get recognition and for the user who might struggle to refer to the exact data when publishing a paper. As it is now a requirement to publish data associated with a paper it is always bets to publish before sharing!

Reviewer: ```suggestion
In the past, researchers often shared data with others before publishing it, even when this data was used for a new publication. This practice should be avoided at any cost. It creates an issue for the creator of the data who won't get recognition, and for the user who might struggle to refer to the exact data when publishing a paper. As it is now required to publish data associated with a paper, it is always best to publish before sharing!
```

Author: I might change the verb tense at the start as this happens all the time not just in the past

```

## Your NCI account

> When you leave, you will keep your NCI credentials IF you keep your contact details up to date through https://my.nci.org.au

Reviewer: ```suggestion
When you leave you can keep your NCI credentials if you keep your contact details up to date through https://my.nci.org.au
```

Make sure to quit all NCI projects you don't need access to anymore. You can be ruthless as it is quite easy to regain access to those projects later on if needed.

> Some projects have strict license terms attached (e.g. access). Your membership might be revoked at any time after you leave the Centre if you haven't negotiated extended access to those projects. Please contact the Lead CI or the project manager to discuss your needs.

Reviewer: ```suggestion
Some projects have strict license terms attached (e.g., ACCESS). Your membership might be revoked any time after you leave the Centre if you haven't negotiated extended access to those projects. Please contact the Lead CI or project manager to discuss your needs.
```

Author: the access project is actually all lower case, I'll commit this and then change it later

**Transferring files**

If you need to transfer files to a different machine (for example from NCI to your University or personal computer):

> use sftp, scp or rsync to transfer files securely ( rsync can be resumed )
> if transferring from/to NCI:
> use the dedicated data-mover nodes if transferring from/to NCI: g-dm.nci.org.au
> use copyq if you want to queue a job

Reviewer: ```suggestion

- use sftp, scp or rsync to transfer files securely (rsync can be resumed)
- if transferring from/to NCI:
    - use the dedicated data-mover nodes if transferring from/to NCI: g-dm.nci.org.au
    - use copyq if you want to queue a job
```

Author: They should all apply to transfer from/to so I'll tidy this up myself after committing
